src/ring_theory/adjoin_root.lean

```diff
@@ -154,4 +197,63 @@ lemma mul_div_root_cancel :
     f.map (of f) :=
 mul_div_eq_iff_is_root.2 $ is_root_root _
 
+section findim
+open vector_space
+open finite_dimensional
+
+/-- The restriction of `adjoin_root.mk f` to the polynomials of degree less than `f`,
+viewed as a isomorphism between vector spaces over `K`. -/
+def degree_lt_linear_equiv (hf : f ≠ 0) : degree_lt K (f.nat_degree) ≃ₗ[K] adjoin_root f :=
```

Comment:
@Vierkantor Do we have the power basis somewhere already? I think you mentioned it at some point.

In general, I think it makes sense to develop that if it isn't there yet (most of it is in this proof, I think).
And then this definition should be something like a 3-liner, using dimensions.

Reply:
I have a couple of power bases, depending on which version of `adjoin` we're talking about. I'll try to get a PR out today.

Reply:
First part of power basis: #4905. If you miss anything, let me know and I'll get to it soon.

Reply:
This has now been merged. @tb65536 Do you think you can use the power basis stuff from #4905 to make this code shorter?

Reply:
@jcommelin Yes, I've pushed the changes.

Reply:
@Vierkantor I generalized the adjoin_root power_basis to not require irreducibility